### PR TITLE
tracing: bump cpp2sky v0.2.1

### PR DESCRIPTION
**Pull request by Shikugawa**
Signed-off-by: Shikugawa <rei@tetrate.io>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/main/PULL_REQUESTS.md)

Commit Message: This PR replaces https://github.com/envoyproxy/envoy/pull/15333/. It upgrades cpp2sky version 0.1.0 -> 0.2.1. It also breaks backward compatibility before SkyWalking 8.3.0 because it uses namespaced-version of data collect protocol
Additional Description:
Risk Level: Low
Testing: Unit
Docs Changes: N/A
Release Notes: Required
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by Shikugawa** [comment]
cc @wbpcode

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -322,26 +322,26 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
         project_url = "https://github.com/apache/skywalking-data-collect-protocol",
         name = "skywalking_data_collect_protocol",
-        sha256 = "fa9ac679624217f30b6e8d5c450365386c610e2d08188a20f0340c3b14401833",
-        urls = ["https://github.com/apache/skywalking-data-collect-protocol/archive/v8.3.0.zip"],
-        strip_prefix = "skywalking-data-collect-protocol-8.3.0",
-        version = "8.3.0",
+        sha256 = "4a3401098e0e49ae56d03cc4890b21c75bab6ccf1c157d73bb0c93fc15bb07c7",
+        urls = ["https://github.com/apache/skywalking-data-collect-protocol/archive/{version}.tar.gz"],
+        strip_prefix = "skywalking-data-collect-protocol-{version}",
+        version = "c987be061e570ce7d0d8f415753f05555e787407",
```
Comment (Shikugawa) [inline bazel/repository_locations.bzl:328]: As I described #15333, cpp2sky depends on not-released feature. This is why we specify SHA directly here.

**Comment by wbpcode** [comment]
cc @moderation

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (moderation) [reply bazel/repository_locations.bzl:328]: 8.4.0 doesn't have the changes you require? https://github.com/apache/skywalking-data-collect-protocol/releases/tag/v8.4.0

Comment (Shikugawa) [reply bazel/repository_locations.bzl]: fixed

**Comment by Shikugawa** [comment]
@moderation PTAL

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -322,26 +322,26 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
-        strip_prefix = "skywalking-data-collect-protocol-8.3.0",
-        version = "8.3.0",
+        sha256 = "edfa970394511213eacc8055b4c13e4e9773e9196122a49e0db68f6162f67dff",
+        urls = ["https://github.com/apache/skywalking-data-collect-protocol/archive/v{version}.tar.gz"],
+        strip_prefix = "skywalking-data-collect-protocol-{version}",
+        version = "8.4.0",
         use_category = ["observability_ext"],
         extensions = ["envoy.tracers.skywalking"],
-        release_date = "2020-11-20",
+        release_date = "2021-01-20",
```
Comment (moderation) [inline bazel/repository_locations.bzl:331]: The UTC release date for this release is `2021-02-01`. I suspect this value works as this is the date of the last commit for the release.

File: bazel/repository_locations.bzl [hunk 3]
```diff
@@ -322,26 +322,26 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
-        sha256 = "a8d870bb4b1c4a05eae319f689d1948927f3f0a5b5fe524db73a4c04121a339a",
-        version = "0.1.1",
+        sha256 = "76117a63cf29355c28a75bc83bd1d7e5bc004039445e7c854ee752dfe66094e6",
+        version = "0.2.1",
         strip_prefix = "cpp2sky-{version}",
         urls = ["https://github.com/SkyAPM/cpp2sky/archive/v{version}.tar.gz"],
         use_category = ["observability_ext"],
         extensions = ["envoy.tracers.skywalking"],
-        release_date = "2021-01-15",
+        release_date = "2021-03-17",
```
Comment (moderation) [inline bazel/repository_locations.bzl:344]: UTC date for this release is `2021-03-24` and I suspect this date currently works as it is the date of the last release

**Review by lizan — APPROVED** [review]

**Pull request merged**